pkg/sdk/sdk.go

```diff
@@ -0,0 +1,154 @@
+package sdk
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	respModel "github.com/PereRohit/util/model"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"time"
+)
+
+type htmlToPdfSvc struct {
+	svcUrl string
+	client http.Client
+}
+
+func NewHtmlToPdfSvc(url string) HtmlToPdfSvcI {
+	return &htmlToPdfSvc{
+		svcUrl: url,
+		client: http.Client{
+			Timeout: 5 * time.Second,
+		},
+	}
+}
+
+type HtmlToPdfSvcI interface {
+	Register([]byte) (string, error)
+	Replace([]byte, string) error
+	GeneratePdf(map[string]interface{}, string) ([]byte, error)
+}
+
+func (h *htmlToPdfSvc) Register(fileBytes []byte) (string, error) {
+	file := bytes.NewReader(fileBytes)
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", "output")
+	if err != nil {
+		return "", err
+	}
+	_, err = io.Copy(part, file)
+	if err != nil {
+		return "", err
+	}
+	writer.Close()
+	r, err := h.client.Post(h.svcUrl+"/v1/register", writer.FormDataContentType(), body)
+	if err != nil {
+		return "", err
+	}
+
+	if r.StatusCode < 200 || r.StatusCode > 299 {
+		return "", fmt.Errorf("non success status code received : %v", r.StatusCode)
+	}
+	resp, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return "", err
+	}
+	var response respModel.Response
+	err = json.Unmarshal(resp, &response)
+	if err != nil {
+		return "", err
+	}
+	x := response.Data
+	m, ok := x.(map[string]interface{})
+	if !ok {
+		errNew := errors.New("unable to parse response data")
+		return "", errNew
+	}
+	if m["id"] == nil {
```

Comment:
Is this how presence of a key in a map is checked?

Reply:
done